### PR TITLE
Migrate to swift part 3: api tester (continued)

### PR DESCRIPTION
**Pull request by aboedo**
Adds more tests to the API Tester

### CODE REVIEW

File: ios/PurchasesHybridCommon/ObjCAPITester/RCHybridCommonAPITest.m [hunk 1]
```diff
@@ -11,17 +11,131 @@
...
+@interface RCCommonFunctionalityAPITest: NSObject
 @end
 
 
-@implementation RCHybridCommonAPITest
+@implementation RCCommonFunctionalityAPITest
 - (void)testAPI {
-    NSString *proxyURL = [RCCommonFunctionality proxyURLString];
-    BOOL simulatesAskToBuyInSandbox = [RCCommonFunctionality simulatesAskToBuyInSandbox];
+    NSString *proxyURL __unused = [RCCommonFunctionality proxyURLString];
```
Comment (aboedo) [inline ios/PurchasesHybridCommon/ObjCAPITester/RCHybridCommonAPITest.m:20]: not much to say for this one, it just calls everything public

File: ios/PurchasesHybridCommon/ObjCAPITester/RCHybridCommonAPITest.m [hunk 2]
```diff
@@ -11,17 +11,131 @@
...
 
+    [RCCommonFunctionality configure];
+    // should issue deprecated warning
+    [RCCommonFunctionality setAllowSharingStoreAccount:NO];
 
+    // should issue deprecated warning
+    [RCCommonFunctionality addAttributionData:@{} network:1 networkUserId:@"asdfg"];
+    
+    [RCCommonFunctionality getProductInfo:@[]
+                           completionBlock:^(NSArray<NSDictionary *> * _Nonnull products) {
```
Comment (vegaro) [inline ios/PurchasesHybridCommon/ObjCAPITester/RCHybridCommonAPITest.m:31]: nit
```suggestion
                          completionBlock:^(NSArray<NSDictionary *> * _Nonnull products) {
```

File: ios/PurchasesHybridCommon/ObjCAPITester/RCHybridCommonAPITest.m [hunk 3]
```diff
@@ -11,17 +11,131 @@
...
+                                                                 RCErrorContainer * _Nullable error) {
+    }];
+
+    [RCCommonFunctionality setAutomaticAppleSearchAdsAttributionCollection:YES];
+    [RCCommonFunctionality getOfferingsWithCompletionBlock:^(NSDictionary * _Nullable offerings,
+                                                             RCErrorContainer * _Nullable error) {
+    }];
+    BOOL isAnonymous __unused = RCCommonFunctionality.isAnonymous;
+
+
```
Comment (vegaro) [inline ios/PurchasesHybridCommon/ObjCAPITester/RCHybridCommonAPITest.m:80]: extra space

Comment (vegaro) [inline ios/PurchasesHybridCommon/ObjCAPITester/RCHybridCommonAPITest.m:80]:
```suggestion

```

File: ios/PurchasesHybridCommon/ObjCAPITester/RCHybridCommonAPITest.m [hunk 4]
```diff
@@ -11,17 +11,131 @@
...
+    [RCCommonFunctionality setFBAnonymousID:@""];
+    [RCCommonFunctionality setMparticleID:@""];
+    [RCCommonFunctionality setOnesignalID:@""];
+    [RCCommonFunctionality setAirshipChannelID:@""];
+    [RCCommonFunctionality setMediaSource:@""];
+    [RCCommonFunctionality setCampaign:@""];
+    [RCCommonFunctionality setAdGroup:@""];
+    [RCCommonFunctionality setAd:@""];
+    [RCCommonFunctionality setKeyword:@""];
+    [RCCommonFunctionality setCreative:@""];
```
Comment (vegaro) [inline ios/PurchasesHybridCommon/ObjCAPITester/RCHybridCommonAPITest.m:136]: All these should also be tested for nullable

**Review by vegaro — CHANGES_REQUESTED** [review]

File: ios/PurchasesHybridCommon/ObjCAPITester/RCHybridCommonAPITest.m (hunk 2 above)
Comment (aboedo) [reply ios/PurchasesHybridCommon/ObjCAPITester/RCHybridCommonAPITest.m:31]: great catch, updating (along with a few other indentation issues)

File: ios/PurchasesHybridCommon/ObjCAPITester/RCHybridCommonAPITest.m (hunk 4 above)
Comment (aboedo) [reply ios/PurchasesHybridCommon/ObjCAPITester/RCHybridCommonAPITest.m:136]: great call, adding

**Comment by aboedo** [comment]
@vegaro this is ready for another pass

**Review by vegaro — APPROVED** [review]

**Pull request merged**